contracts/token-distribution/SaleRounds.sol

```diff
@@ -204,115 +159,186 @@ contract SaleRounds is TokenDistribution, GameOwner, ERC20 {
+    claimableRound(_roundType) {
         RoundType roundType = getRoundTypeByKey(_roundType);
+        require(tokensClaimable, "Tokens are still not claimable");
         require(_msgSender() == _to, "Sender is not a recipient");
 
         ClaimInfo memory claimInfo = ClaimInfo({cliff: roundDistribution[roundType].cliff,
-                                                vesting: roundDistribution[roundType].vesting,
-                                                balance: reservedBalances[roundType][_to],
-                                                claimedBalance: claimedBalances[roundType][_to],
-                                                //granularity can be 30 * 24 * 60 * 60 for monthly or 24 * 60 * 60 for daily? add a field and make constants
-                                                periodGranularity: roundDistribution[roundType].vestingGranularity, //e.g. Days or Months (in seconds!)
-                                                startTime: roundDistribution[roundType].startTime,
-                                                secondsVested: 0,
-                                                vestingForUserPerSecond: 0
-                                            });
+        vesting: roundDistribution[roundType].vesting,
+        balance: reservedBalances[roundType][_to],
+        claimedBalance: claimedBalances[roundType][_to],
+        //granularity can be 30 * 24 * 60 * 60 for monthly or 24 * 60 * 60 for daily? add a field and make constants
+        periodGranularity: roundDistribution[roundType].vestingGranularity, //e.g. Days or Months (in seconds!)
+        startTime: roundDistribution[roundType].startTime,
+        secondsVested: 0,
+        vestingForUserPerSecond: 0
+        });
 
         require(claimInfo.balance > 0, "don't have a reserved balance");
 
         claimInfo.secondsVested = calculateCliffTimeDiff(claimInfo);
-        // abi.encodePacked() method can be used for gas efficiency, campare the gas costs for both usage
+        // abi.encodePacked() method can be used for gas efficiency, compare the gas costs for both usage
         require(claimInfo.secondsVested > 0, string.concat(_roundType, " round is still locked"));
 
         claimInfo.vestingForUserPerSecond = calculateVestingForUserPerSecond(claimInfo);
 
-        uint maximalRelease = getMaximalRelease(claimInfo);
+        uint maximumRelease = getMaximumRelease(claimInfo);
 
-        (uint balanceToRelease, uint unClaimedBalance) = getBalanceToRelease(maximalRelease, claimInfo);
+        (uint balanceToRelease, uint unClaimedBalance) = getBalanceToRelease(maximumRelease, claimInfo);
 
         //Maybe we don't care, but there may be a fractional holding and we want people to be able to just collect that. It's a tiny, tiny value (31.5*10^-12 tokens on 18 decimals)
         //So maybe remove.
         if (claimInfo.vestingForUserPerSecond == 0) balanceToRelease = unClaimedBalance;
```

Comment:
Can we ensure this never becomes 0 by accident (so a user could illegally claim all their tokens at once by creating some wonky chain state?)